modules/custom/az_core/src/Plugin/ConfigProvider/QuickstartConfigProvider.php

```diff
@@ -167,12 +195,9 @@ protected function trimPermissions(array $config) {
    *   An array of permission definitions.
    */
   protected function getPermissionDefinitions() {
-    if (empty($this->permissionDefinitions)) {
-      // @todo Use injection on user.permissions.
-      // @phpstan-ignore-next-line
-      $this->permissionDefinitions = \Drupal::service('user.permissions')->getPermissions();
-    }
-    return $this->permissionDefinitions;
+    // @todo Use injection on user.permissions. and add caching.
```

Comment:
Something seems to be going on with the caching that was here previously. With it enabled, it was receiving the wrong results for some kinds of permissions...investigation needed.

Reply:
> Something seems to be going on with the caching that was here previously. With it enabled, it was receiving the wrong results for some kinds of permissions...investigation needed.

The core patch that adds static caching to the PermissionsHandler service does this in `hook_modules_installed()`:
```
// Reset the user.permissions service to reset statically cached
// permissions.
\Drupal::getContainer()->set('user.permissions', NULL);
```

We may need to do something similar to reset the statically cached permissions in this plugin class, although I'm not sure what the best way to do so would be.